src/components/QuestionTimer.jsx

```diff
@@ -0,0 +1,22 @@
+import { useState, useEffect } from "react";
+
+export default function QuestionTimer({ timeout, onTimeout }) {
+    const [remainingTime, setRemainingTime] = useState(timeout)
+
+    useEffect(() => {
+        setTimeout(onTimeout, timeout)
+    }, [timeout, onTimeout])
+
+    useEffect(() => {
+        setInterval(() => {
+            setRemainingTime(prevRemainingTime => prevRemainingTime - 100)
+        }, 100)
+    }, [])
```

Comment:
Clean up the interval to avoid memory leaks

useEffect(() => {
    const intervalId = setInterval(() => {
        setRemainingTime(prev => prev - 100);
    }, 100);

  // cleanup
    return () => clearInterval(intervalId); 
}, []);